src/content/en/fundamentals/instant-and-offline/service-worker/lifecycle.md

```diff
@@ -0,0 +1,815 @@
+    timeline.to(el.querySelector('.cog-old'), 0.5, {opacity: 0, ease: Quad.easeInOut}, 'takeover');
+    timeline.to(el.querySelector('.cog-new'), 1, {transform: 'matrix(1.1187138,0,0,1.1187138,67.745203,12.407711)', ease: Quint.easeInOut}, 'takeover');
+    timeline.set({}, {}, "+=0.5");
+    timeline.addLabel('page-open');
+    timeline.to(el.querySelector('.controlled'), 0.5, {opacity: 1, ease: Quad.easeInOut}, 'page-open+=0.25');
+    timeline.to(el.querySelector('.diagram-page'), 0.5, {opacity: 1, ease: Quad.easeInOut}, 'page-open');
+
+    if (window.IntersectionObserver) {
+      var observer = new IntersectionObserver(function(changes) {
+        changes.forEach(function(change) {
+          if (change.intersectionRatio) {
+            timeline.play(0);
+            cogRotate.play(0);
+            oldCogRotate.play(0);
+            return;
+          }
+          timeline.pause();
+          cogRotate.pause();
+          oldCogRotate.pause(0);
+        });
+      }, {});
+
+      observer.observe(document.documentElement);
+    }
+    else {
+      timeline.play(0);
+      cogRotate.play(0);
+      oldCogRotate.play(0);
+    }
+  });
+</script>
+{% endframebox %}
+</div>
+</div>
+
+Let's say we changed our service worker script so we responded with a picture of
+a horse rather than a cat:
+
+    const expectedCaches = ['static-v2'];
+
+    self.addEventListener('install', event => {
+      console.log('V2 installing…');
+
+      // cache a horse SVG into a new cache, static-v2
+      event.waitUntil(
+        caches.open('static-v2').then(cache => cache.add('/horse.svg'))
+      );
+    });
+
+    self.addEventListener('activate', event => {
+      // delete any caches that aren't in expectedCaches
+      // which will get rid of static-v1
+      event.waitUntil(
+        caches.keys().then(keys => Promise.all(
+          keys.map(key => {
+            if (!expectedCaches.includes(key)) {
+              return caches.delete(key);
+            }
+          })
+        )).then(() => {
+          console.log('V2 now ready to handle fetches!');
+        })
+      );
+    });
+
+    self.addEventListener('fetch', event => {
+      const url = new URL(event.request.url);
+
+      // serve the cat SVG from the cache if the request is
+      // same-origin and the path is '/dog.svg'
+      if (url.origin == location.origin && url.pathname == '/dog.svg') {
+        event.respondWith(caches.match('/horse.svg'));
+      }
+    });
+
+Note: I have no strong opinions on horses.
+
+[Check out a demo of the
+above](https://cdn.rawgit.com/jakearchibald/80368b84ac1ae8e229fc90b3fe826301/raw/ad55049bee9b11d47f1f7d19a73bf3306d156f43/index-v2.html){:
+.external}, you should still see an image of a cat. Here's why…
+
+### Install
+
+Note that I've changed the cache name from `static-v1` to `static-v2`. This
+means I can set up the new cache without overwriting things in the current one,
+which the old service worker is still using.
```

Comment:
Is there anything to be said about copying assets from the old cache to the new?